### PR TITLE
Type julia_pgd solver

### PR DESCRIPTION
**Pull request by TheoGuyard**
This PR optimizes the julia_pgd solver with typing.

### CODE REVIEW

File: solvers/julia_pgd.jl [hunk 1]
```diff
@@ -1,22 +1,25 @@
...
     L = opnorm(X)^2
-
-    n_features = size(X, 2)
-    w = zeros(n_features, 1)
-    t_new = 1
+    p = size(X, 2)
+    w = zeros(p)
+    r = similar(y)
+    v = similar(w)
+    g = similar(w)
```
Comment (tomMoral) [inline solvers/julia_pgd.jl:17]: Cqn you keep explicit names?

**Review by tomMoral — APPROVED** [review]
Looks good.

Maybe a nitpick but can you keep explicit names?
Also, it would be nice to have a FISTA version, would you be up to code it?

**Comment by TheoGuyard** [comment]
Done 👍🏻

**Comment by mathurinm** [comment]
Thanks @TheoGuyard !

**Pull request merged**